### PR TITLE
Remove npm request and replace with Axios

### PR DESCRIPTION
**Pull request by al-hmcts**
### Jira link (if applicable)

[DFR-3177](https://tools.hmcts.net/jira/browse/DFR-3177)

### Change description ###
The CCD master [pipeline](https://build.hmcts.net/view/FinRem/job/HMCTS_d_to_i/job/finrem-ccd-definitions/job/master/837/pipeline-console/) is failing due to a moderate severity vulnerability in the deprecated HTTP npm Request package.

This package has been [unsupported since 2020](https://github.com/request/request?tab=readme-ov-file#table-of-contents) and should be replaced with a supported equivalent package such as HTTP [Axios](https://www.npmjs.com/package/axios).The CCD master [pipeline](https://build.hmcts.net/view/FinRem/job/HMCTS_d_to_i/job/finrem-ccd-definitions/job/master/837/pipeline-console/) is failing due to a moderate severity vulnerability in the deprecated HTTP npm Request package.

This package has been [unsupported since 2020](https://github.com/request/request?tab=readme-ov-file#table-of-contents) and should be replaced with a supported equivalent package such as HTTP [Axios](https://www.npmjs.com/package/axios).

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

**Comment by Gubbsy** [comment]
Please note: I have not included retries in the HTTP requests as we already have 3 retries on the calling tests, which themselves then have 5 retries defined in Playwright. So I'm happy to bring the number of retries down and fail slightly faster.

**Comment by al-hmcts** [comment]
Please note: I have not included retries in the HTTP requests as we already have 3 retries on the calling tests, which themselves then have 5 retries defined in Playwright. So I'm happy to bring the number of retries down and fail slightly faster.

**Review by hamzahtahirhmcts — APPROVED** [review]

File: test/helpers/utils.js [hunk 1]
```diff
@@ -15,214 +24,168 @@ async function getUserToken(username, password) {
...
+
+  logger.info("Successfully retired user token");
+
+  return authTokenResponse.data.access_token;
 }
 
 async function getUserId(authToken) {
   logger.info('Getting User Id');
 
   const idamBaseUrl = `https://idam-api.${env}.platform.hmcts.net`;
```
Comment (ptrelease) [inline test/helpers/utils.js:55]: @al-hmcts Not your code, but I can see that this and idamCodePath are identicallly declared in a couple of functions.  Could we declare once up by ccdApiUrl.  I like what you've done there better.

File: test/helpers/utils.js [hunk 2]
```diff
@@ -15,214 +24,168 @@ async function getUserToken(username, password) {
...
+
+  const authTokenResponse = await axiosRequest({
+    method: 'post',
+    url: idamBaseUrl + idamAuthPath,
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded'
+    }
+  });
+
+  logger.info("Successfully retired user token");
```
Comment (ptrelease) [inline test/helpers/utils.js:47]: Just checking you don't mean 'retrieved' rather than retired on lines 47 and 35.

File: yarn-audit-known-issues [hunk 3]
```diff
@@ -1 +1 @@
-{"actions":[],"advisories":{},"muted":[],"metadata":{"vulnerabilities":{"info":0,"low":0,"moderate":0,"high":0,"critical":0},"dependencies":828,"devDependencies":0,"optionalDependencies":0,"totalDependencies":828}}
+{"actions":[],"advisories":{},"muted":[],"metadata":{"vulnerabilities":{"info":0,"low":0,"moderate":0,"high":0,"critical":0},"dependencies":828,"devDependencies":0,"optionalDependencies":0,"totalDependencies":828}}
```
Comment (ptrelease) [inline yarn-audit-known-issues:1]: New line

File: test/helpers/utils.js [hunk 4]
```diff
@@ -15,214 +24,168 @@ async function getUserToken(username, password) {
...
+    method: 'get',
+    url: idamBaseUrl + idamDetailsPath,
     headers: { Authorization: `Bearer ${authToken}` }
   });
 
-  logger.debug(JSON.parse(userDetails).id);
-
-  return JSON.parse(userDetails).id;
+  logger.info("Successfully retrieved User ID: ", userDetailsResponse.data.id);
+  return userDetailsResponse.data.id;
```
Comment (ptrelease) [inline test/helpers/utils.js:65]: Just checking line 64 is fine.  Everywhere else you use %s style string interpolation.

Comment (ptrelease) [reply test/helpers/utils.js:65]: (you may have just forgotten the %s)

File: yarn-audit-known-issues (hunk 3 above)
Comment (al-hmcts) [reply yarn-audit-known-issues:1]: @ptrelease, this is an auto-generated file, happy to add in newline but does not come with one by default.

File: test/helpers/utils.js (hunk 1 above)
Comment (al-hmcts) [reply test/helpers/utils.js]: Actioned :)

**Review by ptrelease — APPROVED** [review]

**Review by hamzahtahirhmcts — APPROVED** [review]

**Pull request merged**